control_protocol.md

```diff
@@ -330,17 +330,490 @@ A Component MUST also offer a list of all possibly callable methods in accordanc
 For such a RPC message, the first content frame MUST consist in a JSON-RPC compatible content, for example a single request object or a batch of request objects.
 
 
+### List of methods
+
+List of methods defined by LECO.
+You MAY implement the optional methods of this list and you MUST implement the methods obligatory for your type of Component.
+All methods implemented in a Component MUST adhere to this list or MUST have a name different to any method in this list.
+
+
+#### Component
+
+Any Component, i.e. any participant in the LECO protocol, MUST offer the following methods.
+
+::: json
+{
+    "openrpc": "1.2.6",
+    "info": {
+        "title": "Component",
+        "version": "0.1.0"
+    },
+    "methods": [
+        {
+            "name": "pong",
+            "description": "Respond to a ping.",
+            "params": [],
+            "result": {
+                "name": "result",
+                "schema": {
+                    "type": "null"
+                },
+                "required": true
+            }
+        }
+    ],
+    "components": {
+        "schemas": {}
+    }
+}
+:::
+
+Any Component MAY offer ANY of the following methods.
+Components SHOULD offer ``shut_down``.
+
+::: json
+{
+    "openrpc": "1.2.6",
+    "info": {
+        "title": "Component-optional",
+        "version": "0.1.0"
+    },
+    "methods": [
+        {
+            "name": "set_log_level",
```

Comment:
is there a way to express optionality (`"required": false`) on a method level in openrpc?

Reply:
instead of integers, should we have a categorical/set? info/debug/warning/error/critical? That would probably be more universal, and would more easily map across languages (not everybody uses Python `logging` style).

Reply:
> instead of integers, should we have a categorical/set? info/debug/warning/error/critical? That would probably be more universal, and would more easily map across languages (not everybody uses Python logging style).

you mean "set_info", "set_debug", "set_warning"...?

Reply:
> is there a way to express optionality ("required": false) on a method level in openrpc?

No, as it is not useful, that a server states: I might have optionally that method. Either the server has the method (then it is in the schema) or it does not have the method.

We are basically describing the schema a Component should have, so we have other requirements.

Reply:
> you mean "set_info", "set_debug", "set_warning"...?

No, I mean the parameter of the method
"name": "level",
  "schema": {
  "type": ~~"integer"~~ "one-of-info/debug/warning/error/critical"